ruma-signatures/src/functions.rs

```diff
@@ -642,7 +660,7 @@ pub fn verify_event(
 
 /// Internal implementation detail of the canonical JSON algorithm. Allows customization of the
 /// fields that will be removed before serializing.
-fn canonical_json_with_fields_to_remove(object: &JsonObject, fields: &[&str]) -> String {
+fn canonical_json_with_fields_to_remove(object: &CanonicalJsonObject, fields: &[&str]) -> String {
```

Comment:
About 10 lines down from here `serde_json::to_string` is used, don't we want to use `CanonicalJsonObject::to_canonical_string` to enforce the size limit?